sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -667,4 +667,12 @@ abstract class Catalog {
    * @since 3.5.0
    */
   def listCatalogs(pattern: String): Dataset[CatalogMetadata]
+
```

Comment:
I'm not sure if it's necessary to provide such an interface `def listCatalogs(...)` in `catalog` to align with other similar implementations such as `def listCatalogs(...)`

Reply:
I do not think this is needed for now, so we might be better of not adding.